lib/const.rb

```diff
@@ -70,7 +70,7 @@
          File.file?(File.join(CREW_PREFIX, "etc/crew/meta/glibc_build#{LIBC_VERSION.delete('.')}.filelist")) ? "glibc_build#{LIBC_VERSION.delete('.')}" : ''
        end
     }
-  ].reject!(&:empty?)
```

Comment:
This was handling an edge condition that at some point was causing a crash. Is there any reason to remove it as a safe fallback?

Reply:
Huh!

Reply:
As per the documentation, it will return `nil` if no elements are removed. So, if there aren't any empty elements in `CREW_ESSENTIAL_PACKAGES`, the entire array will be replaced with nil. doesnt seem super ideal to me tbh
<img width="780" height="769" alt="image" src="https://github.com/user-attachments/assets/70dbaf36-188b-42a9-8259-bc2e7b5f6e9c" />

Maybe it would work if we didn't use the in place version? idk what purpose it would really serve though, because only one possible branch gives an empty string, and we can just make it not return the empty string so then theres no need to reject any empty elements because we dont create any in the first place

Reply:
The `%w` is confusing things.
```ruby
irb(main):013> w_array = %w[crew_preload ruby '']
=> ["crew_preload", "ruby", "''"]
irb(main):014> array = ['crew_preload', 'ruby', '']
=> ["crew_preload", "ruby", ""]
irb(main):015> w_array.reject!(&:empty?)
=> nil
irb(main):016> array.reject!(&:empty?)
=> ["crew_preload", "ruby"]
irb(main):017>
```

Reply:
So definitely not intended behavior! But do we actually get `''` in the array, since that with %W gets converted to `"''"`?

Reply:
If I add that back in, I'm not getting breakage on the i686 container:
```
chronos@buppie-i686:2.23 M58 /usr/local/lib/crew/packages (master *>|SPARSE)$ crew const CREW_ESSENTIAL_PACKAGES
CREW_ESSENTIAL_PACKAGES=["bash", "crew_profile_base", "gcc_lib", "gmp", "libxcrypt", "ncurses", "patchelf", "readline", "ruby", "ruby_matrix", "upx", "zlib", "zlib_ng", "zstd", "glibc_build223"]
```

Reply:
(Asking because I specifically added that tiny cleanup to avoid breakage at some point _I was experiencing_.)

Reply:
My container was pre-glibc standalone, so I think it was hitting this path `File.file?(File.join(CREW_PREFIX, "etc/crew/meta/glibc_build#{LIBC_VERSION.delete('.')}.filelist")) ? "glibc_build#{LIBC_VERSION.delete('.')}"`

Regardless, we can just fix it by swapping to `reject` instead of `reject!`, because the non-mutable one is the correct one to use in an assignment

Reply:
let me test that, actually